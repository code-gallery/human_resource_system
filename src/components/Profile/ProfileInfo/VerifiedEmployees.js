import React from 'react'
import PropTypes from 'prop-types'
import _filter from 'lodash/filter'
import _reverse from 'lodash/reverse'
import _take from 'lodash/take'
import Avatar from 'components/Avatar'

const VerifiedEmployees = (props) => {
  const { employees } = props
  const confirmedEmployees = _filter(employees, { confirmed: 1 })
  const last4Employees = _take(_reverse(confirmedEmployees), 4)

  if (confirmedEmployees.length === 0) {
    return null
  }

  return (
    <div className="col-xs-12 col-md-3 hidden-sm hidden-xs organisations-verified-employees">
      <ul className="verified-employees">
        {last4Employees.map((item, index) => {
          return (
            <li key={index}>
              <Avatar
                imageUrl={item.profile_image}
              />
            </li>
          )
        })}
      </ul>
      {confirmedEmployees.length === 1 &&
        <p>1 Verified Employee</p>
      }
      {confirmedEmployees.length > 1 &&
        <p>{confirmedEmployees.length} Verified Employees</p>
      }
    </div>
  )
}

VerifiedEmployees.propTypes = {
  employees: PropTypes.array
}

export default VerifiedEmployees
