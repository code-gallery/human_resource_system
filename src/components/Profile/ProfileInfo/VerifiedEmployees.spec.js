import React from 'react'
import renderer from 'react-test-renderer'
import VerifiedEmployees from './VerifiedEmployees'

describe('components/Profile/.../VerifiedEmployees', () => {
  describe('when props.employees is empty', () => {
    it('renders correctly', () => {
      const props = {
        employees: []
      }
      const tree = renderer.create(
        <VerifiedEmployees {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when there is 1 verified employee', () => {
    it('renders correctly', () => {
      const props = {
        employees: [
          { confirmed: 1, profile_image: null },
          { confirmed: 0, profile_image: null }
        ]
      }
      const tree = renderer.create(
        <VerifiedEmployees {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when there are many verified employees', () => {
    it('renders correctly', () => {
      const props = {
        employees: [
          { confirmed: 1, profile_image: null },
          { confirmed: 1, profile_image: null },
          { confirmed: 1, profile_image: null },
          { confirmed: 0, profile_image: null },
          { confirmed: 1, profile_image: null },
          { confirmed: 1, profile_image: null },
          { confirmed: 1, profile_image: null },
          { confirmed: 0, profile_image: null }
        ]
      }
      const tree = renderer.create(
        <VerifiedEmployees {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
