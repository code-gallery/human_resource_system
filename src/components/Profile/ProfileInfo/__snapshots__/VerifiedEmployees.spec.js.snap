// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Profile/.../VerifiedEmployees when props.employees is empty renders correctly 1`] = `null`;

exports[`components/Profile/.../VerifiedEmployees when there are many verified employees renders correctly 1`] = `
<div
  className="col-xs-12 col-md-3 hidden-sm hidden-xs organisations-verified-employees"
>
  <ul
    className="verified-employees"
  >
    <li>
      <div
        className="Avatar Avatar-white Avatar-size-40"
      >
        <img
          alt="logo"
          src="appii-placeholder.png"
        />
      </div>
    </li>
    <li>
      <div
        className="Avatar Avatar-white Avatar-size-40"
      >
        <img
          alt="logo"
          src="appii-placeholder.png"
        />
      </div>
    </li>
    <li>
      <div
        className="Avatar Avatar-white Avatar-size-40"
      >
        <img
          alt="logo"
          src="appii-placeholder.png"
        />
      </div>
    </li>
    <li>
      <div
        className="Avatar Avatar-white Avatar-size-40"
      >
        <img
          alt="logo"
          src="appii-placeholder.png"
        />
      </div>
    </li>
  </ul>
  <p>
    6
     Verified Employees
  </p>
</div>
`;

exports[`components/Profile/.../VerifiedEmployees when there is 1 verified employee renders correctly 1`] = `
<div
  className="col-xs-12 col-md-3 hidden-sm hidden-xs organisations-verified-employees"
>
  <ul
    className="verified-employees"
  >
    <li>
      <div
        className="Avatar Avatar-white Avatar-size-40"
      >
        <img
          alt="logo"
          src="appii-placeholder.png"
        />
      </div>
    </li>
  </ul>
  <p>
    1 Verified Employee
  </p>
</div>
`;
